File: modules/admin/internal/service/secret/model/secret.go
package model

import "time"

// Secret 秘钥管理表
type Secret struct {
	ID           int       `gorm:"primaryKey;column:c_id;autoIncrement" json:"id"`
	// SpaceID 预留多租户隔离字段，当前为全局共享模式，默认空字符串。
	// 未来如需空间隔离，在 DAO 层的 Create/List/Get/Update/Delete 中读写并过滤此字段。
	SpaceID      string    `gorm:"column:c_space_id;not null;default:''" json:"space_id"`
	SecretID     string    `gorm:"column:c_secret_id;not null" json:"secret_id"`
	Name         string    `gorm:"column:c_name;not null" json:"name"`
	Description  string    `gorm:"column:c_description;not null;default:''" json:"description"`
	Category     string    `gorm:"column:c_category;not null" json:"category"`
	Provider     string    `gorm:"column:c_provider;not null;default:''" json:"provider"`
	SecretType   string    `gorm:"column:c_secret_type;not null;default:'api_key'" json:"secret_type"`
	KeyID        string    `gorm:"column:c_key_id;not null;default:''" json:"key_id"`
	SecretValue  string    `gorm:"column:c_secret_value;not null" json:"secret_value"`
	ExtraConfig  string    `gorm:"column:c_extra_config;not null;default:'{}'" json:"extra_config"`
	Status       string    `gorm:"column:c_status;not null;default:'active'" json:"status"`
	LastUsedAt   *time.Time `gorm:"column:c_last_used_at" json:"last_used_at"`
	LastUsedBy   string    `gorm:"column:c_last_used_by;not null;default:''" json:"last_used_by"`
	Creator      string    `gorm:"column:c_creator;not null;default:''" json:"creator"`
	IsDeleted    string    `gorm:"column:c_is_deleted;not null;default:'false'" json:"is_deleted"`
	CreateTime   time.Time `gorm:"column:c_ctime;type:datetime;default:CURRENT_TIMESTAMP" json:"create_time"`
	ModifyTime   time.Time `gorm:"column:c_mtime;type:datetime;default:CURRENT_TIMESTAMP" json:"modify_time"`
}

func (s *Secret) TableName() string {
	return "t_secrets"
}

File: web/src/api/admin/secret.ts
import { callControl } from './http';

export interface Secret {
  id?: number;
  secret_id?: string;
  name: string;
  description?: string;
  category: string;
  provider?: string;
  secret_type?: string;
  key_id?: string;
  secret_value?: string;
  extra_config?: string;
  status?: string;
  last_used_at?: string;
  last_used_by?: string;
  creator?: string;
  create_time?: string;
  modify_time?: string;
}

export interface ListSecretsReq {
  keyword?: string;
  category?: string;
  provider?: string;
  status?: string;
  offset?: number;
  limit?: number;
}

export interface ListSecretsRsp {
  secrets: Secret[];
  total: number;
}

export interface CreateSecretReq {
  secret: Secret;
}

export interface CreateSecretRsp {
  secret_id: string;
}

export interface UpdateSecretReq {
  secret_id: string;
  name?: string;
  description?: string;
  key_id?: string;
  secret_value?: string;
  extra_config?: string;
  category?: string;
  provider?: string;
  secret_type?: string;
}

export interface ToggleSecretStatusReq {
  secret_id: string;
  status: string;
}

export function listSecrets(req: ListSecretsReq = {}) {
  return callControl<ListSecretsReq, ListSecretsRsp>('secret', 'ListSecrets', req);
}

export function getSecret(secretId: string) {
  return callControl<{ secret_id: string }, { secret: Secret }>('secret', 'GetSecret', { secret_id: secretId });
}

export function createSecret(secret: Secret) {
  return callControl<CreateSecretReq, CreateSecretRsp>('secret', 'CreateSecret', { secret });
}

export function updateSecret(req: UpdateSecretReq) {
  return callControl<UpdateSecretReq, { ret_info?: { code?: number; msg?: string } }>('secret', 'UpdateSecret', req);
}

export function deleteSecret(secretId: string) {
  return callControl<{ secret_id: string }, { ret_info?: { code?: number; msg?: string } }>('secret', 'DeleteSecret', { secret_id: secretId });
}

export function toggleSecretStatus(secretId: string, status: string) {
  return callControl<ToggleSecretStatusReq, { ret_info?: { code?: number; msg?: string } }>('secret', 'ToggleSecretStatus', { secret_id: secretId, status });
}
